src/rules/no-extraneous-dependencies.js

```diff
@@ -31,15 +31,24 @@ function extractDepFields(pkg) {
 
 function getDependencies(context, packageDir) {
   let paths = []
-  try {
-    const packageContent = {
-      dependencies: {},
-      devDependencies: {},
-      optionalDependencies: {},
-      peerDependencies: {},
-      bundledDependencies: [],
-    }
 
+  const packageContent = {
+    dependencies: {},
+    devDependencies: {},
+    optionalDependencies: {},
+    peerDependencies: {},
+    bundledDependencies: [],
+  }
+
+  const haveAddedAnyDependencies = () => [
+    packageContent.dependencies,
+    packageContent.devDependencies,
+    packageContent.optionalDependencies,
+    packageContent.peerDependencies,
+    packageContent.bundledDependencies,
+  ].some(hasKeys)
```

Comment:
```suggestion
  const haveAddedAnyDependencies = [
    packageContent.dependencies,
    packageContent.devDependencies,
    packageContent.optionalDependencies,
    packageContent.peerDependencies,
    packageContent.bundledDependencies,
  ].some(hasKeys);
```
no need to make this a function, it'll be the same result both times.